opencga-analysis/src/main/java/org/opencb/opencga/analysis/customTool/CustomToolExecutor.java

```diff
@@ -36,44 +49,96 @@ protected void check() throws Exception {
 
         // Check any condition
         if (runParams == null) {
-            throw new ToolException("Missing runParams");
+            throw new ToolException("Missing CustomExternalToolParams object.");
+        } else if (StringUtils.isEmpty(runParams.getId())) {
+            throw new ToolException("Missing external tool id.");
+        }
+        Container container = generateDockerObject();
+        checkDockerObject(container);
+    }
+
+    private Container generateDockerObject() throws Exception {
+        OpenCGAResult<ExternalTool> result;
+        if (runParams.getVersion() != null) {
+            Query query = new Query(ExternalToolDBAdaptor.QueryParams.VERSION.key(), runParams.getVersion());
+            result = catalogManager.getExternalToolManager().get(study, Collections.singletonList(runParams.getId()), query,
+                    QueryOptions.empty(), false, token);
+        } else {
+            result = catalogManager.getExternalToolManager().get(study, runParams.getId(), QueryOptions.empty(), token);
         }
-        if (StringUtils.isEmpty(runParams.getCommandLine())) {
-            throw new ToolException("Missing commandLine");
+        if (result.getNumResults() == 0) {
```

Comment:
In case of multiple versions for the same tool, the numResults would still be one, as it would always return only the latest one, right?

Reply:
Yes